### PR TITLE
chore: remove deprecated method

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

File: google-cloud-core/src/test/java/com/google/cloud/BaseWriteChannelTest.java [hunk 1]
```diff
@@ -103,8 +101,12 @@ public void testClose() throws IOException {
...
     channel.close();
-    thrown.expect(ClosedChannelException.class);
-    channel.write(ByteBuffer.allocate(42));
+    try {
+      channel.write(ByteBuffer.allocate(42));
+      Assert.fail();
+    } catch (ClosedChannelException ex) {
+      assertNotNull(ex);
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/BaseWriteChannelTest.java:103]: thrown exception is never null.
I think the test will look much better if you rewrite it as

```
@Test(expected = ClosedChannelException.class)
public void testValidateOpen() throws IOException {
    channel.close();
    channel.write(ByteBuffer.allocate(42));
}
```

File: google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java [hunk 2]
```diff
@@ -188,8 +186,12 @@ public RetryResult afterEval(Exception exception, RetryResult retryResult) {
...
     ExceptionHandler handler = ExceptionHandler.newBuilder().addInterceptors(interceptor).build();
-    thrown.expect(NullPointerException.class);
-    handler.shouldRetry(new Exception(), null);
+    try {
+      handler.shouldRetry(new Exception(), null);
+      Assert.fail();
+    } catch (NullPointerException ex) {
+      assertNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java:188]: I think here you can also use 
`@Test(expected = NullPointerException.class)`
without catching NPE and checking it's message

File: google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java [hunk 3]
```diff
@@ -210,7 +212,11 @@ public RetryResult afterEval(Exception exception, RetryResult retryResult) {
...
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java:209]: `@Test(expected = NullPointerException.class)`

File: google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java [hunk 4]
```diff
@@ -363,8 +360,12 @@ public void testBuilderNoCredentials() {
...
   public void testBuilderNullCredentials() {
-    thrown.expect(NullPointerException.class);
-    TestServiceOptions.newBuilder().setCredentials(null).build();
+    try {
+      TestServiceOptions.newBuilder().setCredentials(null).build();
+      Assert.fail();
+    } catch (NullPointerException ex) {
+      assertNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java:364]: `@Test(expected = NullPointerException.class)`

File: google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java [hunk 5]
```diff
@@ -67,14 +66,22 @@ public static Letter valueOf(String constant) {
...
   public void testNullClass() {
-    expectedException.expect(NullPointerException.class);
-    new StringEnumType<Letter>(null, Letter.CONSTRUCTOR);
+    try {
+      new StringEnumType<Letter>(null, Letter.CONSTRUCTOR);
+      Assert.fail();
+    } catch (NullPointerException ex) {
+      assertNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java:70]: `@Test(expected = NullPointerException.class)`

File: google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java [hunk 6]
```diff
@@ -67,14 +66,22 @@ public static Letter valueOf(String constant) {
...
   public void testNullConstructor() {
-    expectedException.expect(NullPointerException.class);
-    new StringEnumType<Letter>(Letter.class, null);
+    try {
+      new StringEnumType<Letter>(Letter.class, null);
+      Assert.fail();
+    } catch (NullPointerException ex) {
+      assertNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java:80]: `@Test(expected = NullPointerException.class)`

File: google-cloud-core/src/test/java/com/google/cloud/testing/VersionTest.java [hunk 7]
```diff
@@ -41,32 +40,52 @@ public void testFromString() {
...
   public void testFromStringNull() {
-    thrown.expect(NullPointerException.class);
-    Version.fromString(null);
+    try {
+      Version.fromString(null);
+      Assert.fail();
+    } catch (NullPointerException ex) {
+      assertNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/testing/VersionTest.java:89]: `@Test(expected = NullPointerException.class)`

File: google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java [hunk 8]
```diff
@@ -110,8 +117,12 @@ public void testEquals() {
...
   public void testValueOfStrict_invalid() {
-    expectedException.expect(IllegalArgumentException.class);
-    Letter.valueOfStrict("NonExistentLetter");
+    try {
+      Letter.valueOfStrict("NonExistentLetter");
+      Assert.fail();
+    } catch (IllegalArgumentException ex) {
+      assertNotNull(ex.getMessage());
+    }
   }
```
Comment (dmitry-fa) [inline google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java:121]: Here and below in tests for IllegalArgumentException it's better to 
1) Assert.fail("Expected an IllegalArgumentException ")
2) assertThat(ex.getMessage(), is("..."));

**Review by dmitry-fa — CHANGES_REQUESTED** [review]

File: google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java (hunk 5 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java:70]: done

File: google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java (hunk 6 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/StringEnumTest.java:80]: done

File: google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java (hunk 4 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java:364]: done

File: google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java (hunk 3 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java:209]: done

File: google-cloud-core/src/test/java/com/google/cloud/BaseWriteChannelTest.java (hunk 1 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/BaseWriteChannelTest.java:103]: done

File: google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java (hunk 2 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/ExceptionHandlerTest.java:188]: done

File: google-cloud-core/src/test/java/com/google/cloud/testing/VersionTest.java (hunk 7 above)
Comment (athakor) [reply google-cloud-core/src/test/java/com/google/cloud/testing/VersionTest.java:89]: done

**Review by pmakani — APPROVED** [review]
LGTM

**Pull request closed without merging**